src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/CreateFileFromTemplateService.cs

```diff
@@ -34,13 +34,13 @@ public CreateFileFromTemplateService(IUnconfiguredProjectVsServices projectVsSer
         /// Create a file with the given template file and add it to the parent node.
         /// </summary>
         /// <param name="templateFile">The name of the template zip file.</param>
-        /// <param name="parentNode">The node to which the new file will be added.</param>
+        /// <param name="parentDocumentMoniker">The path to the node to which the new file will be added.</param>
         /// <param name="fileName">The name of the file to be created.</param>
         /// <returns>true if file is added successfully.</returns>
-        public async Task<bool> CreateFileAsync(string templateFile, IProjectTree parentNode, string fileName)
+        public async Task<bool> CreateFileAsync(string templateFile, string parentDocumentMoniker, string fileName)
         {
             Requires.NotNull(templateFile, nameof(templateFile));
-            Requires.NotNull(parentNode, nameof(parentNode));
+            Requires.NotNullOrEmpty(parentDocumentMoniker, nameof(parentDocumentMoniker));
```

Comment:
What's the relationship between the unicode standard for whitespace (which is what NotNullOrWhitespace enforces) and paths? This isn't for enforcing "good paths" - this is for finding bugs (you've passed me a path that isn't set "empty and null are interchangeable in .NET apis"). Bad paths will be caught later by other .NET/Win32 APIs .